packages/cube/test/cube/constraintBuilder/DimensionConstraintsBuilder.test.js

```diff
@@ -84,4 +88,23 @@ describe('DimensionConstraintsBuilder', () => {
     it('an integer in between conforms', () => assertConforms(two))
     it('an integer outside the range does not conform', () => assertNotConforms(four))
   })
+  context('built from a string and a cube:Undefined', () => {
+    const builder = new DimensionConstraintsBuilder({ rdf, datatypeParsers })
+    const validator = buildShape(builder, string1, cubeUndefined)
+    const assertConforms = conforms.bind(null, validator)
+    const assertNotConforms = notConforms.bind(null, validator)
+
+    it('the two values conform', () => assertConforms(string1, cubeUndefined))
+    it('another string does not conform', () => assertNotConforms(string2))
+  })
+  context('built from a string and a named node', () => {
```

Comment:
Side note: I think the test file will be more readable if you'd add separation between `context` and "long" `it`

```suggestion
    it('the two values conform', () => assertConforms(string1, cubeUndefined))
    it('another string does not conform', () => assertNotConforms(string2))
  })

  context('built from a string and a named node', () => {
```